lib/Conversion/ATenToTCF/CoreOpConversionPatterns.cpp

```diff
@@ -66,6 +67,68 @@ class ConvertBinaryElementwise : public OpRewritePattern<SourceOp> {
   }
 };
 
+/// The ATen Conv2dOp has seven arguments:
+///   input, weight, bias, stride, padding, dilation, groups
+
+class ConvertATenConv2d : public OpRewritePattern<aten::Conv2dOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(aten::Conv2dOp srcConv2dOp,
+                                PatternRewriter &rewriter) const override {
+    auto results = srcConv2dOp.getOperation()->getResults();
+    assert(srcConv2dOp.getNumOperands() == 7 && "expected seven (7) operands");
+    assert(results.size() == 1 && "expected single result op");
+    // TODO: Replace constant int-list constraints for stride, padding, and dilation; and, constant int constraint for groups.
+    auto *strideOp = srcConv2dOp.stride().getDefiningOp();
+    assert(strideOp->getNumOperands() == 2 && "expected stride length of 2");
+    auto *strideOperand0Op = strideOp->getOperand(0).getDefiningOp();
+    auto *strideOperand1Op = strideOp->getOperand(1).getDefiningOp();
+    if (!matchPattern(strideOp, m_Op<Basicpy::BuildListOp>())
+      || !matchPattern(strideOperand0Op, m_One())
+      || !matchPattern(strideOperand1Op, m_One())
+      ) {
+      return rewriter.notifyMatchFailure(
+          srcConv2dOp, "aten.conv2d to tcf.conv_2d_nchw currently only supports stride == [1, 1]");
+    }
+    auto *paddingOp = srcConv2dOp.padding().getDefiningOp();
+    assert(paddingOp->getNumOperands() == 2 && "expected padding length of 2");
+    auto *paddingOperand0Op = paddingOp->getOperand(0).getDefiningOp();
+    auto *paddingOperand1Op = paddingOp->getOperand(1).getDefiningOp();
+    if (!matchPattern(paddingOp, m_Op<Basicpy::BuildListOp>())
+      || !matchPattern(paddingOperand0Op, m_Zero())
+      || !matchPattern(paddingOperand1Op, m_Zero())
+      ) {
+      return rewriter.notifyMatchFailure(
+          srcConv2dOp, "aten.conv2d to tcf.conv_2d_nchw currently only supports padding == [0, 0]");
+    }
+    auto *dilationOp = srcConv2dOp.dilation().getDefiningOp();
+    assert(dilationOp->getNumOperands() == 2 && "expected dilation length of 2");
```

Comment:
not all basicpy.build_list ops have 2 operands. You need to check this and `return failure()` in that case, otherwise the compiler can crash on some user-provided inputs.

Reply:
Concretely I would do

```
auto paddingList = srcConv2dOp.padding().getDefiningOp<Basicpy::BuildListOp>();
if (!paddingList || paddingList->getNumOperands() != 2)
  return rewriter.notifyMatchfailure(...);
... stuff with matchPattern ...
```